smack-extensions/src/main/java/org/jivesoftware/smackx/avatar/AvatarMetadataStore.java

```diff
@@ -0,0 +1,46 @@
+/**
+ *
+ * Copyright 2019 Paul Schaub
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.avatar;
+
+import org.jxmpp.jid.EntityBareJid;
+
+/**
+ * The {@link AvatarMetadataStore} interface defines methods used by the {@link UserAvatarManager} to determine,
+ * whether the client already has a local copy of a published avatar or if the user needs to be informed about the
+ * update in order to download the image.
+ */
+public interface AvatarMetadataStore {
```

Comment:
I am confused by this "store", which does not actually seem to store avatars, just their availability.

Why not make it a real store-interface, e.g.
```
interface AvatarMetadataStore {
   Avatar getAvatar(EntityBareJid jid, String itemId);
   putAvatar(EntityBareJid jid, String itemId, Avatar avatar);
```
where Avatar is a meta class that encapsulates the MIME type and the avatar bytes.